### PR TITLE
Code cleanup/refactoring

### PR DESCRIPTION
**Pull request by akhillanger**
This is some generic cleanup/refactoring that is also prerequisite for the upcoming pull requests on persistent collectives and FCNAME/FUNCNAME macro removal.

### CODE REVIEW

**Comment by akhillanger** [comment]
test:jenkins/ch4/most
test:jenkins/ch3/most

**Comment by akhillanger** [comment]
@raffenet are these test results ok? I cannot reproduce the ch4/ofi spawintrax failure locally and @wesbland told to ignore the ch3/sock/freebsd spawn test failure.

**Comment by raffenet** [comment]
@akhillanger it's possible that spawn failure is just intermittent. Hopefully with Gengbin's patches it will be solved permanently. In any case, I will re-run the `ch4/ofi` test for good measure.
test:jenkins/ch4/ofi

**Comment by akhillanger** [comment]
`ch4/ofi` tests have passed. This PR was reviewed internally by Wesley. @raffenet it can be reviewed now. I will soon open the persistent collectives pull request.

**Comment by akhillanger** [comment]
@raffenet Can we merge this. I have a few pull requests that are based on this.

File: src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90 [hunk 1]
```diff
@@ -493,15 +493,15 @@ function MPIR_Ibarrier_c(comm, request) &
     integer(c_int) :: ierror
 end function MPIR_Ibarrier_c
 
-function MPIR_Op_commutative_c(op, commute) &
+function MPIR_Op_ptr_commutative_c(op, commute) &
```
Comment (raffenet) [inline src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90]: I'm not sure this makes things any clearer, IMO. I would prefer to drop it.

File: src/include/mpir_op.h [hunk 2]
```diff
@@ -113,10 +113,10 @@ extern MPIR_Op MPIR_Op_builtin[MPIR_OP_N_BUILTIN];
 extern MPIR_Op MPIR_Op_direct[];
 extern MPIR_Object_alloc_t MPIR_Op_mem;
 
-#define MPIR_Op_add_ref(_op) \
-    do { MPIR_Object_add_ref(_op); } while (0)
-#define MPIR_Op_release_ref(_op, _inuse) \
-    do { MPIR_Object_release_ref(_op, _inuse); } while (0)
+#define MPIR_Op_add_ref(op_p_) \
+    do { MPIR_Object_add_ref(op_p_); } while (0)
+#define MPIR_Op_ptr_release_ref(op_p_, inuse_) \
```
Comment (raffenet) [inline src/include/mpir_op.h]: This changes a macro name as well as variable name. Both are inconsistent with other object ref macros, such as `MPIR_Comm_`, `MPIR_Request_`. I would prefer to drop this for now, or do a more broad cleanup than just the Op and Datatype macros in the future.

File: src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90 (hunk 1 above)
Comment (akhillanger) [reply src/binding/fortran/use_mpi_f08/mpi_c_interface_nobuf.f90]: I have dropped this commit.

File: src/include/mpir_op.h (hunk 2 above)
Comment (akhillanger) [reply src/include/mpir_op.h]: This commit is only changing variable name. I have applied the same change to `MPIR_Comm` and `MPIR_Request` structures.

File: test/mpi/coll/testlist.in [hunk 3]
```diff
@@ -35,7 +35,7 @@ bcasttest 8
 bcasttest 10
 bcast_full 4 timeLimit=300
 # More that 8 processes are required to get bcast to switch to the long
-# msg algorithm (see coll definitions in mpiimpl.h)
+# msg algorithm
```
[the remark this answers is not in the document]
Comment (akhillanger) [reply test/mpi/coll/testlist.in]: Done

File: test/mpi/topo/Makefile.am [hunk 4]
```diff
@@ -18,17 +18,16 @@ noinst_PROGRAMS = \
     cartshift1    \
     cartsuball    \
     cartcreates   \
+    dgraph_unwgt  \
```
[the remark this answers is not in the document]
Comment (akhillanger) [reply test/mpi/topo/Makefile.am]: Ordering changes removed.

File: src/mpid/ch3/include/mpidimpl.h [hunk 5]
```diff
@@ -969,7 +969,7 @@ extern char *MPIDI_DBG_parent_str;
     }
 #endif
 
-/* This is used to quote a name in a definition (see FUNCNAME/FCNAME below) */
+/* This is used to quote a name in a definition (see FCNAME before function definitions) */
```
[the remark this answers is not in the document]
Comment (akhillanger) [reply src/mpid/ch3/include/mpidimpl.h]: Comment dropped.

**Comment by akhillanger** [comment]
@raffenet I have addressed your comments. I also added one commit, the last one.

File: src/include/mpir_op.h (hunk 2 above)
Comment (raffenet) [reply src/include/mpir_op.h]: This still changes `MPIR_Op_release_ref` -> `MPIR_Op_ptr_release_ref`.

Comment (akhillanger) [reply src/include/mpir_op.h]: Oops. Fixed it now.

**Comment by raffenet** [comment]
OK, let me get a baseline test result, then I will finish the review.
test:jenkins/ch3/tcp
test:jenkins/ch4/ofi

**Comment by akhillanger** [comment]
@raffenet The updates are just some very simple style cleanup with Wesley's improved pre-commit hook.

**Comment by raffenet** [comment]
This looked mostly OK. I rebased it to latest master and made some minor edits to the commit messages, but nothing else. Going to run the full set of tests now.
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by raffenet** [comment]
Whitespace failure can be ignored. It is not from changes in this PR.

**Pull request merged**